### DISCUSSION TITLE
Cannot import micropip in locally running pyodide console

### DISCUSSION THREAD

**Discussion by Vuizur** [pyodide/pyodide#4109] [Q&A] [answered]
I run a pyodide server according to the "Creating a pyodide package" tutorial with the command `python -m http.server --directory ./dist`
(This runs inside the docker container inside WSL2).
I access this from Firefox from Windows and when I try to import micropip, I get:
```
Welcome to the Pyodide terminal emulator 🐍
Python 3.11.1 (main, Jan 25 2023 19:17:40) on WebAssembly/Emscripten
Type "help", "copyright", "credits" or "license" for more information.
>>> import micropip
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/lib/python3.11/site-packages/micropip/__init__.py", line 1, in <module>
    from ._micropip import _list as list
  File "/lib/python3.11/site-packages/micropip/_micropip.py", line 22, in <module>
    from packaging.markers import default_environment
  File "/lib/python3.11/site-packages/packaging/markers.py", line 11, in <module>
    from pyparsing import (  # noqa: N817
ModuleNotFoundError: The module 'pyparsing' is included in the Pyodide distribution, but it is not installed.
You can install it by calling:
  await micropip.install("pyparsing") in Python, or
  await pyodide.loadPackage("pyparsing") in JavaScript
See https://pyodide.org/en/stable/usage/loading-packages.html for more details.
```

The server log is as following:

```
172.17.0.1 - - [31/Aug/2023 10:10:47] "GET /console.html HTTP/1.1" 200 -
172.17.0.1 - - [31/Aug/2023 10:10:47] "GET /pyodide.js HTTP/1.1" 200 -
172.17.0.1 - - [31/Aug/2023 10:10:47] "GET /pyodide.asm.js HTTP/1.1" 200 -
172.17.0.1 - - [31/Aug/2023 10:10:47] "GET /pyodide_py.tar HTTP/1.1" 200 -
172.17.0.1 - - [31/Aug/2023 10:10:47] code 404, message File not found
172.17.0.1 - - [31/Aug/2023 10:10:47] "GET /favicon.ico HTTP/1.1" 404 -
172.17.0.1 - - [31/Aug/2023 10:10:47] "GET /pyodide.asm.data HTTP/1.1" 200 -
172.17.0.1 - - [31/Aug/2023 10:10:47] "GET /repodata.json HTTP/1.1" 200 -
172.17.0.1 - - [31/Aug/2023 10:10:47] "GET /pyodide.asm.wasm HTTP/1.1" 200 -
172.17.0.1 - - [31/Aug/2023 10:10:52] "GET /pyparsing-3.0.9-py3-none-any.whl HTTP/1.1" 200 -
172.17.0.1 - - [31/Aug/2023 10:10:52] "GET /micropip-0.2.0-py3-none-any.whl HTTP/1.1" 200 -
172.17.0.1 - - [31/Aug/2023 10:10:52] "GET /packaging-21.3-py3-none-any.whl HTTP/1.1" 200 -
```

Importing micropip works without problems here: https://pyodide.org/en/stable/console.html

What am I missing? Thanks for the help!

**Comment by ryanking13** [comment]
Hmm, I'm not sure what the problem is based on the logs you showed me - what version of Pyodide are you using? If you use the latest version of Pyodide (0.23.4), do you get the same issue?

**Reply by Vuizur** [reply]
I run it from source (from my branch, which is basically up to date with master: https://github.com/Vuizur/pyodide/tree/add-spacy). 
Inspired from your question I remembered that I pulled a few commits from master since following the tutorial, so I rebuilt emsdk using:
```
make -C emsdk
make -C cpython
```
and rebuilt spacy, but I still have the same error.
Should I test some other version?

**Reply by ryanking13** [reply]
The log shows that you have `pyodide_py.tar` in your build artifacts, which have been removed after Python 0.23.0. So I think you are building an old version of Pyodide for some reason.

**Reply by Vuizur** [reply, marked as answer]
Thanks a lot for the help! I finally found the reason: I think it was related to the fact that even after rerunning `make` and everything I still had the old wheel for `packaging` next to the new wheel (or maybe I forgot to properly refresh my browser windows, but at least deleting the old wheel worked).
